MediaBrowser.MediaEncoding/Attachments/AttachmentExtractor.cs

```diff
@@ -230,13 +231,166 @@ public async Task<(MediaAttachment Attachment, Stream Stream)> GetAttachment(Bas
+                {
+                    @lock.Dispose();
+                }
+            }
+        }
+
+        private async Task CacheAllAttachmentsInternal(
+            string mediaPath,
+            string inputFile,
+            MediaSourceInfo mediaSource,
+            List<int> extractableAttachmentIds,
+            CancellationToken cancellationToken)
+        {
+            var outputPaths = new List<string>();
+            var processArgs = string.Empty;
+
+            foreach (var attachmentId in extractableAttachmentIds)
+            {
+                var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, attachmentId);
+
+                Directory.CreateDirectory(Path.GetDirectoryName(outputPath) ?? throw new FileNotFoundException($"Calculated path ({outputPath}) is not valid."));
+
+                outputPaths.Add(outputPath);
+                processArgs += string.Format(
+                    CultureInfo.InvariantCulture,
+                    " -dump_attachment:{0} \"{1}\"",
+                    attachmentId,
+                    EncodingUtils.NormalizePath(outputPath));
+            }
+
+            processArgs += string.Format(
+                CultureInfo.InvariantCulture,
+                " -i \"{0}\" -t 0 -f null null",
+                inputFile);
+
+            int exitCode;
+
+            using (var process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        Arguments = processArgs,
+                        FileName = _mediaEncoder.EncoderPath,
+                        UseShellExecute = false,
+                        CreateNoWindow = true,
+                        WindowStyle = ProcessWindowStyle.Hidden,
+                        ErrorDialog = false
+                    },
+                    EnableRaisingEvents = true
+                })
+            {
+                _logger.LogInformation("{File} {Arguments}", process.StartInfo.FileName, process.StartInfo.Arguments);
+
+                process.Start();
+
+                try
+                {
+                    await process.WaitForExitAsync(cancellationToken).ConfigureAwait(false);
+                    exitCode = process.ExitCode;
+                }
+                catch (OperationCanceledException)
+                {
+                    process.Kill(true);
+                    exitCode = -1;
+                }
+            }
+
+            var failed = false;
+
+            if (exitCode == -1)
+            {
+                failed = true;
+
+                foreach (var outputPath in outputPaths)
+                {
+                    try
+                    {
+                        _logger.LogWarning("Deleting extracted media attachment due to failure: {Path}", outputPath);
+                        _fileSystem.DeleteFile(outputPath);
+                    }
+                    catch (FileNotFoundException)
+                    {
```

Comment:
Dont drop exceptions without logging and explanation

Reply:
I don't think we should notify the user about this case. `ffmpeg` failed, so it is normal that one or more expected output files do not exist. Added a comment to the code stating this, but if you would like me to log a warning anyways, sure, I'll do that, just let me know what the message should be.